src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,566 @@
+        expected = new int[][]{{0, 1, 4}, {0, 1, 2}};
+        verifyActionResult(expected, "filterQuery", query);
+
+        // id
+        expected = new int[][]{{2}};
+        query = "id:id02   ";
+        verifyActionResult(expected, "testdata", "true", "filterQuery", query);
+    }
+
+    private void testFiltersCombination() {
+        Date today = TimeHelper.getDateOffsetToCurrentTime(0);
+        Date twoDaysAgo = TimeHelper.getDateOffsetToCurrentTime(-2);
+
+        // filterQuery with showing all URI
+        int[][] expected = new int[][]{{0, 3, 5, 7, 8, 9}};
+        String query = "info:keyword1";
+        verifyActionResult(expected, "filterQuery", query, "all", "true");
+
+        // another filterQuery with showing all URI
+        expected = new int[][]{{0, 1, 3, 7, 8, 9}, {0, 1, 4}, {3}};
+        query = String.format("person:Name1 | from:%s and to:%s",
+                              formatterAdminTime.format(twoDaysAgo), formatterAdminTime.format(today));
+        verifyActionResult(expected, "filterQuery", query, "all", "true");
+
+        // filterQuery with showing test data
+        expected = new int[][]{{0, 1, 2}};
+        query = "role:Admin";
+        verifyActionResult(expected, "filterQuery", query, "testdata", "true");
+
+        // filterQuery with showing everything
+        expected = new int[][]{{2, 4, 5, 6, 7}, {1, 3, 4}, {0, 1, 2}};
+        query = String.format("time:50 | from:%s and to:%s",
+                              formatterAdminTime.format(twoDaysAgo), formatterAdminTime.format(today));
+        verifyActionResult(expected, "filterQuery", query, "testdata", "true", "all", "true");
+    }
+
+    private void testLogMessageInDifferentVersions() {
+        // version query is controlled by GAE itself
+        // so there is no need to write comprehensive test case for it
+
+        int[][] expected = new int[][]{{}};
+        String query = "version:2";
+        verifyActionResult(expected, "filterQuery", query);
+
+        expected = new int[][]{{0, 1, 3, 4, 5}};
+        query = "version:2, 1";
+        verifyActionResult(expected, "filterQuery", query);
+    }
+
+    private void testStatusMessage() {
+        Date yesterday = TimeHelper.getDateOffsetToCurrentTime(-1);
+
+        // test statusMessage for default search
+        AdminActivityLogPageAction action = getAction();
+        String statusMessage = getShowPageResult(action).getStatusMessage();
+        verifyStatusMessage(statusMessage, 12, 5, yesterday);
+        verifyLocalTimeInStatusMessage(statusMessage, yesterday, Const.SystemParams.ADMIN_TIME_ZONE_DOUBLE);
+
+        // test statusMessage with filterQuery
+        String query = "person:idOfInstructor1OfCourse1";
+        action = getAction("filterQuery", query);
+        statusMessage = getShowPageResult(action).getStatusMessage();
+        verifyStatusMessage(statusMessage, 12, 1, yesterday);
+        // TODO: fix the bug
```

Comment:
Do make the issue right after this is merged, otherwise we're adding more technical debts.